aerogear/mobile-developer-console
# Services JS SDK docs generated based on bound services on the Configuration tab

## Motivation
https://issues.jboss.org/browse/AEROGEAR-8032
https://issues.jboss.org/browse/AEROGEAR-8035
https://issues.jboss.org/browse/AEROGEAR-8036

## What
Information about AeroGear services configuration is generated based on bound services in the Configuration tab.

## Why
To make it easy for developers to set up their project.

## How
Mobile app information contains information about types of bound services. There is documentation stored in `ui/src/components/configuration/sdk-config-docs`, planned to support more frameworks in future.

## Verification Steps
Add the steps required to check this change. Following an example.
 
1. Go to some mobile app
2. On the *Configruation* tab, you should see generated documentation
3. Try to create new service binding or delete existing one
4. Documentation should update based on the bound services

## Checklist:


## Progress

- [x] Implementation for Cordova docs
- [x] Add other supported frameworks?
- [x] Update view based on flats
- [x] SDK services docs links
- [x] SDK services docs links with AEROGEAR_DOCS_VERSION env. variable
- [x] Add tests for some components used

## Additional Notes
Cordova is now hardcoded, it needs to be updated with info about what frameworks will be supported.
![image](https://user-images.githubusercontent.com/2912228/49370517-a970df80-f6f4-11e8-9623-0c97f1cb1d8e.png)
